keyvault/client/secrets/README.md

```diff
@@ -0,0 +1,252 @@
+        .build();
+Secret secret = secretClient.setSecret("secret_name", "secret_value").value();
+DeletedSecret deletedSecret = client.deleteSecret("secret_name").value();
+System.out.printf("Deleted Secret's deletion time %s", deletedSecret.deletedDate().toString());
+```
+
+## Async API Examples
+
+The following sections provide several code snippets covering some of the most common Azure Key Vault Secret Service tasks, including:
+- [Create a Secret](#create-a-Secret)
+- [Retrieve a Secret](#retrieve-a-Secret)
+- [Update an existing Secret](#update-an-existing-Secret)
+- [Delete a Secret](#delete-a-Secret)
+
+### Create a Secret
+
+Create a Secret to be stored in the Azure Key Vault. There are two ways to store a Secret:
+- setSecret creates a new secret in the key vault. if the secret with name already exists then a new version of the secret is created.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+
+secretAsyncClient.setSecret("secret_name", "secret_value").subscribe(secretResponse ->
+  System.out.printf("Secret is created with name %s and value %s \n", secretResponse.value().name(), secretResponse.value().value()));
+```
+
+### Retrieve a Secret
+
+Retrieve a previously stored Secret by calling getSecret.
+```Java
+SecretClient secretAsyncClient = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+        
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse ->
+  System.out.printf("Secret with name %s , value %s \n", secretResponse.value().name(),
+  secretResponse.value().value()));
+```
+
+### Update an existing Secret
+
+Update an existing Secret by calling updateSecret.
+```Java
+SecretAsyncClient secretAsyncClient = SecretAsyncClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+
+secretAsyncClient.getSecret("secretName").subscribe(secretResponse -> {
+     Secret secret = secretResponse.value();
+     //Update the expiry time of the secret.
+     secret.expires(OffsetDateTime.now().plusDays(50));
+     secretAsyncClient.updateSecret(secret).subscribe(secretResponse ->
+         System.out.printf("Secret's updated not before time %s \n", secretResponse.value().notBefore().toString()));
+   });
+```
+
+### Delete a Secret
+
+Delete an existing Secret by calling deleteSecret.
+```Java
+SecretClient client = SecretClient.builder()
+        .endpoint("https://myvault.vault.azure.net")
+        .credentials(AzureCredential.DEFAULT)
+        .build();
+        
+secretAsyncClient.deleteSecret("secretName").subscribe(deletedSecretResponse ->
+   System.out.printf("Deleted Secret's deletion time %s \n", deletedSecretResponse.value().deletedDate().toString()));
+```
+
+## Troubleshooting
+
+### General
```

Comment:
an example on how to handle the exception or how it is returned might be worth adding

Reply:
Agreed, check out Sameeksha's version - also does Java have logging?